tests/test_convolution_1d.py

```diff
@@ -0,0 +1,194 @@
+                                         [[np.zeros([1, 2, 4]), np.zeros([1, 1, 1, 1])]] +
+                                         [[np.zeros([1, 2, 4, 5]), np.zeros([1, 1, 1, 1, 1])]]
+    )
+    def test_check_dim(self, iterable, filter_type):
+        raise_exception = any(trial.ndim != 3 for trial in iterable)
+        if raise_exception:
+            with pytest.raises(ValueError, match="conv_trials must be an iterable of 3D arrays"):
+                utils.nan_pad_conv(iterable, 3, filter_type)
+        else:
+            utils.nan_pad_conv(iterable, 3, filter_type)
+
+    @pytest.mark.parametrize("filter_type", ["causal", "acausal", "anti-causal", ""])
+    @pytest.mark.parametrize("iterable",
+                             [[np.zeros([2, 4, 5]), np.zeros([1, 1, 10])]]
+                             )
+    def test_conv_type(self, iterable, filter_type):
+        raise_exception = not (filter_type in ["causal", "anti-causal", "acausal"])
+        if raise_exception:
+            with pytest.raises(ValueError, match='filter_type must be "causal", "acausal"'):
+                utils.nan_pad_conv(iterable, 3, filter_type)
+        else:
+            utils.nan_pad_conv(iterable, 3, filter_type)
+
+    @pytest.mark.parametrize("iterable",
+                             [[np.zeros([2, 4, 5]), np.zeros([2, 4, 6])]]
+                             )
+    @pytest.mark.parametrize("window_size", [0, 1, 2, 3, 5, 6])
+    def test_padding_nan_causal(self, window_size, iterable):
+        raise_exception = (not isinstance(window_size, int)) or (window_size <= 0)
+        if raise_exception:
+            with pytest.raises(ValueError, match="window_size must be a positive integer!"):
+                utils.nan_pad_conv(iterable, window_size, "anti-causal")
+        else:
+            padded = utils.nan_pad_conv(iterable, window_size, "causal")
+            for trial in padded:
+                print(trial.shape, window_size)
+            assert all(np.isnan(trial[:window_size]).all() for trial in padded), "Missing NaNs at the " \
+                                                                                       "beginning of the array!"
+            assert all(not np.isnan(trial[window_size:]).any() for trial in padded), "Fund NaNs at the " \
+                                                                                      "end of the array!"
+            assert all(padded[k].shape[0] == iterable[k].shape[0] - 1 + window_size for k in range(len(padded))), \
+                "Size after padding doesn't match expectation. Should be T + window_size - 1."
+
+    @pytest.mark.parametrize("iterable",
+                             [[np.zeros([2, 5, 4]), np.zeros([2, 6, 4])]]
+                             )
+    @pytest.mark.parametrize("window_size", [0, 1, 2, 3, 5, 6])
+    def test_padding_nan_anti_causal(self, window_size, iterable):
+        raise_exception = (not isinstance(window_size, int)) or (window_size <= 0)
```

Comment:
not actually testing what happens if `window_size` is not an `int`, probably should

Reply:
good catch